src/metabase/lib/legacy_metric.cljc

```diff
@@ -112,15 +112,19 @@
                                        (keep-indexed (fn [index aggregation-clause]
                                                        (when (lib.util/clause-of-type? aggregation-clause :metric)
                                                          [(get aggregation-clause 2) index])))
-                                       (lib.aggregation/aggregations query stage-number))]
-         (cond
-           (empty? metrics)             nil
-           (empty? metric-aggregations) (vec metrics)
-           :else                        (mapv (fn [metric-metadata]
-                                                (let [aggregation-pos (-> metric-metadata :id metric-aggregations)]
-                                                  (cond-> metric-metadata
-                                                    aggregation-pos (assoc :aggregation-position aggregation-pos))))
-                                              metrics)))))))
+                                       (lib.aggregation/aggregations query stage-number))
+             results (cond
+                       (empty? metrics)             nil
+                       (empty? metric-aggregations) (vec metrics)
+                       :else                        (mapv (fn [metric-metadata]
+                                                            (let [aggregation-pos (-> metric-metadata :id metric-aggregations)]
+                                                              (cond-> metric-metadata
+                                                                aggregation-pos (assoc :aggregation-position aggregation-pos))))
+                                                          metrics))
+             sorted (sort-by (some-fn :display-name :name) results)]
```

Comment:
By use of `sort-by`, `results` become seq instead of a vector. Can you confirm no call site expects vector? Ie. for example no operations as `(what-used-to-be-a-vector <index>)` are performed?